assets/js/llms-metabox-voucher.js

```diff
@@ -166,7 +166,7 @@
 })( jQuery );
 
 function llms_on_voucher_duplicate (results) {
-	if (results.length) {
+	if ( undefined !== results && results.length) {
```

Comment:
that's good, but this is the callback of an ajax call made to check if there are voucher code duplicates.
I'd rather avoid to do the ajax call at all if there are no codes, what do you think?

Reply:
I think we need to make the AJAX callback because of the `else` statement. The else clause submits the form if there are no duplicates.

Suppose, I've added another course and then clicked update without any code, I still need to submit the form to save the data.
What I can do is, move the `else` clause to the ajax callback. What do you say?
But that'll still be the same process.

Reply:
Of course you have to submit the form :D 
I just said we don't need to do an ajax call to check for duplicates when there are no voucher codes :D 
E.g. returning `true` in the click handler if there's not voucher code?!

Reply:
Hey @eri-trabiccolo

I've updated the code according to your suggestions. Have a look!